tests/cpp/test_multidevice_transformer.cpp

```diff
@@ -278,29 +278,65 @@ std::vector<TensorView*> mlp(
     TensorView* b0,
     TensorView* w1,
     TensorView* b1,
-    const DeviceMesh& mesh) {
+    const DeviceMesh& mesh,
+    bool sequence_parallel = false,
+    TensorView* mask = nullptr) {
   const DataType dtype = w0->dtype();
-  // // Linear 0
+
+  if (sequence_parallel) {
+    // Input arrives sharded and must be allgathered back
+    x->setDeviceMesh(mesh);
+    x->axis(0)->parallelize(ParallelType::DIDx);
+    x = set(x); // allgather
+    x->setDeviceMesh(mesh);
+    x->axis(0)->parallelize(ParallelType::Serial);
+    // Reshape back to 2D for linearOp
```

Comment:
is there a plan to patch LinearOp to accept nD tensors ?

Reply:
Good question, the ATen interface can accept an arbitrary number of input dimensions for activations. @Priya2698 would know better. 

I might be mixing up the restriction with weights and activations...

Reply:
LinearOp supports nD input activations. >2D weights however will be squeezed and results will be unsqueezed -- preceding dimensions are treated as device dimensions. According to the code [here](https://github.com/NVIDIA/Fuser/blob/628a47e33ac418dc7b2a3a79a01f52174c2fe325/csrc/ir/nodes.cpp#L4379-L4394), I think it should work out of the box without "reshaping back to 2D".

Reply:
Yes. Activations can have any number of dimensions. Weights can be 1D or 2D.
@wujingyue removed 1D weights for ease of DID implementation of linear.